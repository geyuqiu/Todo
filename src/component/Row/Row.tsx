import React from "react";

import {TableRow} from '../Table/TableRow';
import {TableCell} from '../Table/TableCell';
import {Todo} from './model';

type RowProps = {
	todo: Todo
};

export const Row = ({todo: todo}: RowProps) => {
	return (
		<TableRow className="text-base" border>

			<TableCell>
				{todo.id}
			</TableCell>

			<TableCell>
				{todo.title}
			</TableCell>

			<TableCell>
				{String(todo.completed)}
			</TableCell>

		</TableRow>
	);
}
