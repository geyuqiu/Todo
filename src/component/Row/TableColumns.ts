export const tableColumns = [
	{
		Header: '#',
		accessor: 'id'
	},
	{
		Header: 'Title',
		accessor: 'title',
		className: "text-right"
	},
	{
		Header: 'Completed',
		accessor: 'completed',
	}
];
