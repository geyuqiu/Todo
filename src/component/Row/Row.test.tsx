import {render} from "@testing-library/react";
import React from "react";

import {Row} from "./Row";
import {Todo} from './model';

describe("Row", () => {
	it("should match snapshot", () => {
		const todo = {
			id: 1,
			title: "delectus aut autem",
			completed: false
		} as Todo;

		const { container } = render(
				<Row todo={todo}/>
		);

		expect(container).toMatchSnapshot();
	});
});
