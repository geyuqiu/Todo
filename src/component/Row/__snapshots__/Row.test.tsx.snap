// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should match snapshot 1`] = `
<div>
  <tr
    class="TableRow__TableRowStyled-sc-990v33-0 iJGVRV"
    data-testid="TableRow"
  >
    <td
      class="border-gray-lightest border"
    >
      <div
        class="TableCell__TableCellInnerWrapper-l6uokn-0 BHpwz"
      >
        1
      </div>
    </td>
    <td
      class="border-gray-lightest border"
    >
      <div
        class="TableCell__TableCellInnerWrapper-l6uokn-0 BHpwz"
      >
        delectus aut autem
      </div>
    </td>
    <td
      class="border-gray-lightest border"
    >
      <div
        class="TableCell__TableCellInnerWrapper-l6uokn-0 BHpwz"
      >
        false
      </div>
    </td>
  </tr>
</div>
`;
